### ISSUE TITLE
I18Manager is not forcing RTL on app launch but works when restarting the app.

### ISSUE THREAD

**Issue by husseinAbdulnabi** [facebook/react-native#32509]
### Description

using ```I18nManager.forceRTL``` is not forcing whatever boolean value I give on app launch and require the app to restart in-order to apply the direction, this wasn't happening on 0.64.1

### Version

0.66.1

### Output of `react-native info`

System:
    OS: macOS 12.0.1
    CPU: (8) arm64 Apple M1
    Memory: 242.44 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.3.0 - /opt/homebrew/bin/node
    Yarn: 1.22.10 - /opt/homebrew/bin/yarn
    npm: 7.23.0 - /opt/homebrew/bin/npm
    Watchman: 2021.06.07.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 29.0.2, 30.0.2, 30.0.3
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.2 => 0.66.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

add the code below anywhere in `app.js`
```
    I18nManager.allowRTL(true);
    I18nManager.forceRTL(true);
    I18nManager.swapLeftAndRightInRTL(true);
```
- expected outcome:
the app to launch in RTL layout

- actual outcome: 
the app is not changing the layout to RTL

**Comment by ghoudi01** [comment]
Any updates

**Comment by mvrty** [comment]
Seeing same problem.

When using RNRestart to reload the app it respects RTL, but when closing the app down then re-open it doesn't work.

**Comment by mvrty** [comment]
Can confirm https://github.com/facebook/react-native/commit/fcead14b0effe2176a5d08ad50ee71e48528ddbd is changing this behaviour. Is there some workaround for this or is this the expected behaviour going forward? cc @Saadnajmi

**Comment by Saadnajmi** [comment]
Yes, my last change to il8n manager does cause that behavior. We can revert it if need be. What scenario did you need to force RTL for, out of curiosity?

**Comment by husseinAbdulnabi** [comment]
> Yes, my last change to il8n manager does cause that behavior. We can revert it if need be. What scenario did you need to force RTL for, out of curiosity?

@Saadnajmi we have support for arabic language in our app thus the app must be running in RTL mode so no matter what the scenario is, the user must always see the app in RTL layout from the very beginning of the app lifecycle, the change is forcing the app to launch in LTR mode and requires a restart to flip it back to RTL after the initial launch, which is absolutely not an option in our case, restarting sometimes causes crashes, we are holding off the upgrade for our app for now until this gets resolved

**Comment by mvrty** [comment]
Thanks for replying @Saadnajmi ! We offer an app in several languages (some LTR, some RTL). After user has updated its preferences regarding language, we of course respect that on app launch. With current behavior app will start in LTR mode even if an RTL language is selected. Since it’s not RTL only we can’t force RTL in AppDelegate for example.

Reverted back to RN 0.65.2 but hopeful we can get previous behavior back in coming releases.

**Comment by Saadnajmi** [comment]
I see. As an Arabic speaker I understand. I thought that if the app launched with an RTL language, RN would launch with RTL without needing il8nManager to force it. I'm kind of surprised that's not the behavior. I'll try to send a PR reverting my change. Apologies for the mismatch in behavior!

**Comment by husseinAbdulnabi** [comment]
@Saadnajmi thank you!
